code/modules/mob/living/simple_animal/hostile/grue.dm

```diff
@@ -0,0 +1,591 @@
+	if(ismoulting)
+		desc="[desc] This one seems dead and lifeless."
+	else
+		playsound(src, 'sound/misc/grue_screech.ogg', 50, 1)
+	..()
+
+/mob/living/simple_animal/hostile/grue/attack_animal(mob/living/simple_animal/M)
+	if(M==src) //Prevent the grue from attacking itself, might help avoid misclicks while attempting to smash lights.
+		return
+	else
+//		if(prob(20)&&lifestage>1)
+//			playsound(src, 'sound/misc/grue_growl.ogg', 50, 1) //occasionally growl while attacking
+		M.unarmed_attack_mob(src)
+
+
+//Reproduction via egglaying.
+/mob/living/simple_animal/hostile/grue/verb/reproduce()
+	set name = "Reproduce"
+	set desc = "Spawn offspring in the form of an egg."
+	set category = "Grue"
+	if (!config.grue_egglaying) //Check if egglaying is enabled.
+		return
+	if (lifestage==3) //must be adult
+		if (eatencharge<=0)
+			to_chat(src, "<span class='notice'>You need to feed more first.</span>")
+			return
+		else if (!isturf(loc))
+			to_chat(src, "<span class='notice'>You need more room to reproduce.</span>")
+			return
+		else if (stat==UNCONSCIOUS)
+			to_chat(src, "<span class='notice'>You must be awake to reproduce.</span>")
+			return
+		else if (busy)
+			to_chat(src, "<span class='notice'>You are already doing something.</span>")
+			return
+		else
+			handle_reproduce()
+
+	else
+		to_chat(src, "<span class='notice'>You haven't grown enough to reproduce yet.</span>")
+
+/mob/living/simple_animal/hostile/grue/proc/handle_reproduce()
+
+	if(eatencharge>=1)
+		busy=1
+		to_chat(src, "<span class='notice'>You start to push out an egg...</span>")
+		visible_message("<span class='warning'>[src] tightens up...</span>")
+		if(do_after(src, src, 5 SECONDS))
+			to_chat(src, "<span class='notice'>You lay an egg.</span>")
+			visible_message("<span class='warning'>[src] pushes out an egg!</span>")
+			eatencharge--
+//			playsound(T, 'sound/effects/splat.ogg', 50, 1)
+//			var/mob/living/simple_animal/grue_egg/new_grueegg = new(get_turf(src))
+			new /mob/living/simple_animal/grue_egg(get_turf(src))
+		busy=0
+
+	else
+		to_chat(src, "You need to feed more first.")
+		return
+
+//Procs for grabbing players.
+/mob/living/simple_animal/hostile/grue/proc/request_player()
+	var/list/candidates=list()
+	for(var/mob/dead/observer/G in get_active_candidates(ROLE_GRUE, poll="Would you like to become a grue?"))
+		if(!G.client)
+			//testing("Client of [G] inexistent")
+			continue
+
+		//#warn Uncomment me.
+		/*if(G.client.holder)
+			//testing("Client of [G] is admin.")
+			continue*/
+
+		if(isantagbanned(G))
+			//testing("[G] is jobbanned.")
+			continue
+
+		candidates += G
+
+	if(!candidates.len)
+		//message_admins("Unable to find a mind for [src.name]")
+		return 0
+
+	shuffle(candidates)
+	for(var/mob/i in candidates)
+		if(!i || !i.client)
+			continue //Dont bother removing them from the list since we only grab one wizard
+		return i
+
+	return 0
+
+/mob/living/simple_animal/hostile/grue/proc/transfer_personality(var/client/candidate)
```

Comment:
I am not sure if this is needed. Maybe the antag role `post_mind_transfer` is a better place for this.

Reply:
I'll check that out.